### PR TITLE
feat: support multiple roles for one user

### PR DESCRIPTION
**Pull request by tyyzqmf**
----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

## Summary

(describe what this merge request does)

## Implementation highlights

(describe how the merge request does for feature changes, share the RFC link if it has)

## Test checklist

- [x] add new test cases
- [ ] all code changes are covered by unit tests
- [ ] end-to-end tests
  - [x] deploy web console with CloudFront + S3 + API gateway
  - [ ] deploy web console within VPC
  - [ ] deploy ingestion server
    - [ ] with MSK sink
    - [ ] with KDS sink
    - [ ] with S3 sink
  - [ ] deploy data processing
  - [ ] deploy data modeling
    - [ ] new Redshift Serverless
    - [ ] provisioned Redshift
    - [ ] Athena
  - [ ] deploy with reporting

## Is it a breaking change

- [ ] add parameters without default value in stack
- [ ] introduce new service permission in stack
- [ ] introduce new top level stack module

## Miscellaneous

- [ ] introduce new symbol link source file(s) to be shared among infra code, web console frontend, and web console backend

### CODE REVIEW

File: frontend/src/pages/user/UserList.tsx [hunk 1]
```diff
@@ -136,15 +158,15 @@ const UserList: React.FC = () => {
...
       header: t('user:labels.tableColumnRole'),
-      minWidth: 200,
+      minWidth: 240,
       editConfig: {
         editingCell: (item: IUserTableItem, { setValue, currentValue }: any) =>
           renderEditRoleCell(item, setValue, currentValue),
       },
-      cell: (e: { role: string }) => {
-        return getRoleName(e.role);
+      cell: (e: { roles: IUserRole[] }) => {
```
Comment (chenhaiyun) [inline frontend/src/pages/user/UserList.tsx:168]: Update the **e** type to `IUserTableItem`

File: frontend/public/locales/zh-CN/user.json [hunk 2]
```diff
@@ -19,8 +19,11 @@
     "deleteTip1": "是否确实要删除该用户 ",
     "deleteTip2": " 删除用户可能导致用户无法登录。",
     "userSetting": "用户设置",
+    "selectUserRolePlaceholder": "请选择角色",
     "settingUserRoleJsonPath": "获取用户角色的JsonPath",
     "settingUserRoleJsonPathDesc": "从id_token中获取用户角色或者组信息的JsonPath，例如：$.realm_access.roles",
+    "settingUserAdminRoleNames": "管理员角色名称",
+    "settingUserAdminRoleNamesDesc": "OIDC中的角色映射为Clickstream中的管理员，多个角色名称用逗号分隔，例如：operator,others",
```
Comment (zxkane) [inline frontend/public/locales/zh-CN/user.json:26]: operator -> admin

File: frontend/src/ts/utils.ts [hunk 3]
```diff
@@ -405,3 +406,41 @@ export const getLocaleLngDescription = (description: {
...
+    (role) => role === IUserRole.ANALYST || role === IUserRole.ADMIN
+  );
+};
+
+export const isOperatorRole = (roles: IUserRole[] | undefined) => {
+  if (!roles) {
+    return false;
+  }
+  return (
+    roles.some((role) => role === IUserRole.OPERATOR) && !isAdminRole(roles)
```
Comment (zxkane) [inline frontend/src/ts/utils.ts:444]: Why exclude admin role?

**Review by zxkane — CHANGES_REQUESTED** [review]

File: frontend/src/ts/utils.ts (hunk 3 above)
Comment (tyyzqmf) [reply frontend/src/ts/utils.ts]: Useless method, deleted.

File: frontend/public/locales/zh-CN/user.json (hunk 2 above)
Comment (tyyzqmf) [reply frontend/public/locales/zh-CN/user.json]: Modified

File: frontend/src/pages/user/UserList.tsx (hunk 1 above)
Comment (tyyzqmf) [reply frontend/src/pages/user/UserList.tsx]: Modified.

**Review by chenhaiyun — APPROVED** [review]

**Review by zxkane — APPROVED** [review]

**Pull request merged**